kfs-core/src/main/java/edu/arizona/kfs/fp/service/impl/PaymentMethodGeneralLedgerPendingEntryServiceImpl.java

```diff
@@ -0,0 +1,455 @@
+        if ( actualTotalsByChart == null ) {
+            actualTotalsByChart = getNonOffsetActualTotalsByChart(document);
+        }
+
+        for ( String chart : actualTotalsByChart.keySet() ) {
+            KualiDecimal offsetAmount = actualTotalsByChart.get(chart);
+            if ( !KualiDecimal.ZERO.equals(offsetAmount) ) {
+                PaymentMethodChart pmc = pm.getPaymentMethodChartInfo(chart, new java.sql.Date( document.getDocumentHeader().getWorkflowDocument().getDateCreated().toDate().getTime() ) );
+                if ( pmc == null ) {
+                    LOG.warn( "No Applicable PaymentMethodChart found for chart: " + chart + " and date: " + document.getDocumentHeader().getWorkflowDocument().getDateCreated() );
+                    // skip this line - still attempt for other charts
+                    continue;
+                }
+                String clearingChartCode = pmc.getClearingChartOfAccountsCode();
+                String clearingAccountNumber = pmc.getClearingAccountNumber();
+                String clearingObjectCode = pmc.getClearingFinancialObjectCode(); // liability object code
+                
+                GeneralLedgerPendingEntry apOffsetEntry = new GeneralLedgerPendingEntry(document.getGeneralLedgerPendingEntry(0));
+                apOffsetEntry.setTransactionLedgerEntrySequenceNumber(new Integer(sequenceHelper.getSequenceCounter()));
+    
+                apOffsetEntry.setChartOfAccountsCode(clearingChartCode);
+                apOffsetEntry.setAccountNumber(clearingAccountNumber);
+                apOffsetEntry.setFinancialObjectCode(clearingObjectCode);
+                apOffsetEntry.setFinancialSubObjectCode(GENERAL_LEDGER_PENDING_ENTRY_CODE.getBlankFinancialSubObjectCode());
+                apOffsetEntry.setSubAccountNumber(GENERAL_LEDGER_PENDING_ENTRY_CODE.getBlankSubAccountNumber());
+                apOffsetEntry.setProjectCode(GENERAL_LEDGER_PENDING_ENTRY_CODE.getBlankProjectCode());
+    
+                // retrieve object type
+                ObjectCode objectCode = getObjectCodeService().getByPrimaryIdForCurrentYear(clearingChartCode, clearingObjectCode);
+                if ( objectCode == null ) {
+                    LOG.fatal("Specified offset object code: " + clearingChartCode + "-" + clearingObjectCode + " does not exist - failed to generate CC offset entries", new RuntimeException() );
+                    return false;
+                }
+                apOffsetEntry.setFinancialObjectTypeCode(objectCode.getFinancialObjectTypeCode());
+                apOffsetEntry.setTransactionLedgerEntryAmount(offsetAmount.abs());
+                apOffsetEntry.setTransactionDebitCreditCode(offsetAmount.isNegative()?KFSConstants.GL_DEBIT_CODE:KFSConstants.GL_CREDIT_CODE);
+                apOffsetEntry.setFinancialBalanceTypeCode(KFSConstants.BALANCE_TYPE_ACTUAL);
```

Comment:
This constant has been deprecated. Should be getting this from SystemOptions. See LaborBalanceSummaryReportServiceImpl.getActualBalanceTypes(Integer fiscalYear).